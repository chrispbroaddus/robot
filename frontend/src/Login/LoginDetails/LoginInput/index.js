import React from 'react'
import PropTypes from 'prop-types'
import styled from 'styled-components'

const LoginField = styled.input`
  width: 210px;
  height: 30px;
  padding: 5px 10px;
  margin: 0 auto;
  margin-bottom: 10px;
  border: 1px solid #cfd8dc;
  border-radius: 5px;
  color: #90a4ae;
  font-size: 0.8rem;
  display: block;

  &::placeholder {
    color: #90a4ae;
    font-family: Montserrat-Light;
    letter-spacing: 0.5px;
  }
`

const LoginInput = props => {
  return (
    <LoginField
      name={props.name}
      type={props.type}
      value={props.value}
      onChange={props.handleChange}
      placeholder={props.placeholder}
      innerRef={props.refFunc}
    />
  )
}

LoginInput.propTypes = {
  value: PropTypes.string,
  placeholder: PropTypes.string,
  handleChange: PropTypes.func.isRequired,
  name: PropTypes.string,
  type: PropTypes.string,
  // Using the property "refFunc" because React complains when we try to use ref
  refFunc: PropTypes.func
}

LoginInput.defaultProps = {
  type: 'text'
}

export default LoginInput
