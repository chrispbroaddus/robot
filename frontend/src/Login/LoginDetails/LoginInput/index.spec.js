import React from 'react'
import { shallow, mount } from 'enzyme'
import toJson from 'enzyme-to-json'
import LoginInput from './index'

describe('LoginInput', () => {
  let props

  beforeEach(() => {
    props = {
      value: '',
      handleChange: () => {},
      placeholder: ''
    }
  })

  it('should render self and subcomponents', () => {
    const wrapper = shallow(<LoginInput {...props} />)
    expect(toJson(wrapper)).toMatchSnapshot()
  })

  it('should render props correctly', () => {
    props.value = 'Test'
    props.handleChange = jest.fn()
    props.placeholder = 'Test placeholder'
    const wrapper = mount(<LoginInput {...props} />)
    const subject = wrapper.find('input')
    expect(subject.prop('value')).toEqual(props.value)
    expect(subject.prop('placeholder')).toEqual(props.placeholder)

    subject.simulate('change', { target: { value: 'Test description' } })
    expect(props.handleChange).toBeCalled()
  })
})
