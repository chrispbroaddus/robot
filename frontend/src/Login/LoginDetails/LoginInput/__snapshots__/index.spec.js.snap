// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LoginInput should render self and subcomponents 1`] = `
<styled.input
  onChange={[Function]}
  placeholder=""
  type="text"
  value=""
/>
`;
